camunda-sdk-java/java-common/src/main/java/io/camunda/common/auth/SelfManagedAuthentication.java

```diff
@@ -69,17 +69,21 @@ public Map.Entry<String, String> getTokenHeader(Product product) {
     if (tokens.containsKey(product)) {
       token = tokens.get(product);
     } else {
-      Identity identity = identityConfig.get(product).getIdentity();
-      String audience = jwtConfig.getProduct(product).getAudience();
-      Tokens identityTokens = identity.authentication().requestToken(audience);
-      try {
-        identity.authentication().verifyToken(identityTokens.getAccessToken());
-      } catch (TokenExpiredException exception) {
-        identityTokens = identity.authentication().renewToken(identityTokens.getRefreshToken());
-      }
-      tokens.put(product, identityTokens.getAccessToken());
-      token = tokens.get(product);
+      token = getIdentityToken(product);
     }
     return new AbstractMap.SimpleEntry<>("Authorization", "Bearer " + token);
   }
+
+  private String getIdentityToken(Product product) {
+    Identity identity = identityConfig.get(product).getIdentity();
+    String audience = jwtConfig.getProduct(product).getAudience();
+    Tokens identityTokens = identity.authentication().requestToken(audience);
+    try {
+      identity.authentication().verifyToken(identityTokens.getAccessToken());
+    } catch (TokenExpiredException exception) {
+      identityTokens = identity.authentication().renewToken(identityTokens.getRefreshToken());
+    }
+    tokens.put(product, identityTokens.getAccessToken());
+    return tokens.get(product);
```

Comment:
I would recommend to only handle the token creation here and put this part of the coder in the caller that is responsible for managing the tokens.

Reply:
I agree, the function is doing multiple things at once. Refactored [here](https://github.com/camunda-community-hub/spring-zeebe/pull/606/commits/305d96c91b8840fc082ccca94f3550bb6ad1b912)